src/core/api_client/apiClient.ts

```diff
@@ -16,22 +16,50 @@ import type {
 } from './types';
 import type { OidcSimpleClient } from '~core/auth/OidcSimpleClient';
 
+type EventMap = {
+  error: ApiClientError;
+  poolUpdate: Map<string, string>;
+  idle: boolean;
+};
+
+type EventType = keyof EventMap;
+type Listener<T> = (arg: T) => void;
+type ListenerMap = {
+  [K in EventType]: Set<Listener<EventMap[K]>>;
+};
+
 export class ApiClient {
-  private listeners = new Map([['error', new Set<(e: ApiClientError) => void>()]]);
+  private listeners: ListenerMap = {
+    error: new Set(),
+    poolUpdate: new Set(),
+    idle: new Set(),
+  };
   private baseURL!: string;
+  private requestPool = new Map<string, string>();
```

Comment:
NIT: Is `requestPool` only used for logging/listeners? The name gave me the idea that it might be used in queueing or something. Maybe a comment or more specific name would be nice

Reply:
will do on next iteration